internal/csi-addons/rbd/replication.go

```diff
@@ -326,7 +326,23 @@ func (rs *ReplicationServer) DisableVolumeReplication(ctx context.Context,
 	case librbd.MirrorImageDisabling:
 		return nil, status.Errorf(codes.Aborted, "%s is in disabling state", volumeID)
 	case librbd.MirrorImageEnabled:
-		return corerbd.DisableVolumeReplication(rbdVol, mirroringInfo, force)
+		err = rbdVol.DisableVolumeReplication(mirroringInfo, force)
+		if err == nil {
+			return &replication.DisableVolumeReplicationResponse{}, nil
+		}
+		switch {
+		case strings.Contains(err.Error(), "failed to get local state"):
+			return nil, status.Error(codes.Internal, err.Error())
+		case strings.Contains(err.Error(), "secondary image status is"):
+			return nil, status.Error(codes.InvalidArgument, err.Error())
+		case strings.Contains(err.Error(), "failed to disable image mirroring"):
+			return nil, status.Error(codes.Internal, err.Error())
+		case strings.Contains(err.Error(), "failed to get mirroring info of image"):
+			return nil, status.Error(codes.Internal, err.Error())
+		case strings.Contains(err.Error(), "in disabling state"):
+			return nil, status.Error(codes.Aborted, err.Error())
+		}
```

Comment:
You should define an error and return it and use `errors.IS/As` for comparison instead of string comparison.